File: src-tauri/src/gfx_api.rs
use crate::launch::LaunchMethodInterface;

#[tauri::command]
pub fn get_gfx_apis(launch_method: LaunchMethodInterface) -> Vec<String> {
  match launch_method {
    LaunchMethodInterface::Windows => vec!["dx11".to_string(), "vulkan".to_string()],
    LaunchMethodInterface::WindowsLauncher => vec!["dx11".to_string(), "vulkan".to_string()],
  }
}

File: src/scripts/GfxApi.ts
import { LaunchMethod } from "./LaunchMethod";

export enum GfxApi {
  Dx11 = 'dx11',
  Vulkan = 'vk',
}

export function getGfxApis(method: LaunchMethod): GfxApi[] {
  switch (method) {
    case LaunchMethod.Windows:
      return [GfxApi.Dx11, GfxApi.Vulkan];
    case LaunchMethod.Linux:
      return [GfxApi.Vulkan];
    case LaunchMethod.MacGPTK:
      return [GfxApi.Dx11];
    default:
      return [];
  }
}
